src/transaction.ts

```diff
@@ -993,6 +1021,16 @@ export class Transaction implements TransactionStorageStructure {
       if (txnForEnc.apas !== undefined) {
         txn.appForeignAssets = txnForEnc.apas;
       }
+      if (txnForEnc.apbx !== undefined) {
+        txn.boxes = txnForEnc.apbx.map((box) => ({
+          // Translate foreign app index to app ID
+          appIndex:
+            box.i !== 0 || box.i !== txn.appIndex
```

Comment:
What's the reason for checking `box.i !== txn.appIndex`? Won't the conditional always evaluate to true since it is not possible for `box.i` to be both `0` and `txn.appIndex` at the same time?

Reply:
Good catch, should be `&&`

I'll change the condition so it is also easier to read by using the positive form.